packages/eslint-plugin/rules/jsx-a11y-anchor-has-content.js

```diff
@@ -0,0 +1,173 @@
+	let text = false;
+	let args = [];
+	if ( node.type === 'CallExpression' ) {
+		const translateFunctionName = getTranslateFunctionName( node.callee );
+		// get text if we are going through a translate function
+		args = getTranslateFunctionArgs(
+			translateFunctionName,
+			node.arguments,
+			false
+		).map( getTextContentFromNode );
+		const hasUnknownFunction = args.filter( Boolean ).length === 0;
+		// An unknown function call in a translate function may produce a valid string
+		// but verifying it is not straight-forward, so we keep `text` as false to indicate we should bail
+		if ( ! hasUnknownFunction ) {
+			// we think we have a valid string in args, so we join them
+			text = args.join( '' );
+		}
+	} else {
+		// if this `getTextContentFromNode` doesn't find a text it'll return false, which is what we want when we need to bail
+		text = getTextContentFromNode( node );
+	}
+	return text;
+};
+/**
+ * A wrapper around eslint-plugin/anchor-has-content to allow usage with `createInterpolateElement`
+ *
+ * @param {*} context https://eslint.org/docs/latest/extend/custom-rules#the-context-object
+ * @return {Object} the rule object
+ */
+const rule = function ( context ) {
+	return {
+		JSXOpeningElement: ( node ) => {
+			/**
+			 * Uses the custom components rules to determine if the node is an anchor or not.
+			 */
+			const options = context.options[ 0 ] || {};
+			const componentOptions = options.components || [];
+			const typeCheck = [ 'a' ].concat( componentOptions );
+			if (
+				// in theory we should never actually have a falsy value as a custom component option, but we do `&& type` just in case it happens
+				typeCheck.filter(
+					( type ) => type === node?.name?.name && type
+				).length === 0
+			) {
+				// bail - we only care about anchors and custom components configured through rules
+				return;
+			}
+
+			const interpolatedNode = context
+				.getAncestors()
+				.find(
+					( n ) =>
+						n.type === 'CallExpression' &&
+						n.callee.name === 'createInterpolateElement'
+				);
+
+			if ( ! interpolatedNode ) {
+				// wrap - not interpolated, carry on with the normal a11y anchor-has-content rule
+				return anchorRule.create( context ).JSXOpeningElement( node );
+			}
+			// createInterpolateElement's 1st argument, which is the text to be interpolated
+			// we may get a CallExpression, e.g: __('some text'), or a Literal e.g: 'some text'
+			const textParam = interpolatedNode.arguments[ 0 ] ?? [];
+			const interpolatedText = getInterpolatedNodeText( textParam );
+			// bail if we can't get a string value from the first param of `createInterpolateElement`
+			if ( false === interpolatedText ) {
+				return;
+			}
+			try {
+				validateAnchorsContent( interpolatedText, typeCheck );
+			} catch ( e ) {
+				// if the error is not what we expect, something else went wrong and we should re-throw
+				if ( ! meta.messages.hasOwnProperty( e.message ) ) {
+					throw e;
```

Comment:
The only other thing in `validateAnchorsContent` that can fail is the constructor of `RegExp`, but I'm wondering if we should really re-throw here, rather than report? Mainly curious how throwing compares to `context.report` in terms of developer experience, as throwing means that we completely stop the linting process.

I'm thinking something along the lines of:

```javascript
const meta = {
...
  messages: {
    unexpected: "Unexpected error occurred: {{error}}"
...
}
```

And then:

```suggestion
				context.report( {
					node: interpolatedNode,
					messageId: meta.messages.unexpected,
					data: { error: e.message },
				} );
```

Not sure if returning (silent failing) is another option, but we would probably want to report such errors.

Reply:
Like you, I really don't think this should _ever_ happen, so with the re-throw, my intention was "If this ever happened, I'd prefer someone tells us about it rather than think it's a problem in their code.". 

The downside as we chatted about today is that yea, if it happens, it will stop the linter which is also not ideal. 

@tyxla do you have any thoughts on this decision? We could be more explicit in the report (and not rethrow) or add some hint about sending an issue, etc.

Reply:
Yeah, I like Boro's idea 👍  I think if we just throw it might create a false expectation that the consumer has an issue in their code (or 3rd party code that they're using), while if we report it (with some additional context if possible), it can be the most helpful to them, bringing immediate value.

Reply:
I do expect this shouldn't happen, but there are no downsides to having it handled just in case 😉

Reply:
thank you! we just got it to report instead of throw and added a little nudge in the message :)